### PR TITLE
Appcred authtype

### PR DESCRIPTION
**Pull request by scrungus**
we add the auth_type to the application credential secret for `capi-janitor-system`, as if it is not provided then upon cluster deletion we see: 
```
Traceback (most recent call last):
  File "/venv/lib/python3.10/site-packages/capi_janitor/openstack/operator.py", line 266, in wrapper
    result = await handler(**kwargs)
  File "/venv/lib/python3.10/site-packages/capi_janitor/openstack/operator.py", line 381, in on_openstackcluster_event
    await purge_openstack_resources(
  File "/venv/lib/python3.10/site-packages/capi_janitor/openstack/operator.py", line 152, in purge_openstack_resources
    async with openstack.Cloud.from_clouds(clouds, cacert = cacert) as cloud:
  File "/venv/lib/python3.10/site-packages/capi_janitor/openstack/openstack.py", line 219, in from_clouds
    if config["auth_type"] != "v3applicationcredential":
KeyError: 'auth_type'
```

### CODE REVIEW

File: magnum_capi_helm/driver_utils.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def sanitized_name(name, suffix=None):
         "[^a-z0-9]+",
         "-",
         (f"{name}-{suffix}" if suffix else name).lower(),
-    )
+    ).strip("-")
```
Comment (JohnGarbutt) [inline magnum_capi_helm/driver_utils.py:37]: Why is this needed?

File: magnum_capi_helm/driver.py [hunk 2]
```diff
@@ -691,7 +691,7 @@ def _storageclass_definitions(self, context, cluster):
 
         for volume_type in volume_types:
             storage_class = {
-                "name": volume_type,
+                "name": driver_utils.sanitized_name(volume_type),
```
Comment (JohnGarbutt) [inline magnum_capi_helm/driver.py:694]: Isn't this your other unrelated change?

File: magnum_capi_helm/tests/common/test_app_creds.py [hunk 3]
```diff
@@ -67,6 +67,7 @@ def test_create_app_cred(self, mock_client):
                         "application_credential_secret": "pass",
                         "auth_url": "http://keystone",
                     },
+		    "auth_type" : "v3applicationcredential",
```
Comment (JohnGarbutt) [inline magnum_capi_helm/tests/common/test_app_creds.py:70]: I think the indentation isn't quite right here? The auto format should fix that if you just run "tox"

**Review by JohnGarbutt — DISMISSED** [review]

File: magnum_capi_helm/driver.py (hunk 2 above)
Comment (scrungus) [reply magnum_capi_helm/driver.py]: yeah sorry, was layered on top as I was testing on tav1. fixed now

**Review by mkjpryor — APPROVED** [review]
LGTM
